lib/src/import_path_base.dart

```diff
@@ -0,0 +1,283 @@
+
+    return tree;
+  }
+
+  /// Performs the imports search and returns the tree.
+  /// - If [dots] is `true` uses the `dots` style for the tree.
+  /// - See [ASCIIArtTree].
+  Future<ASCIIArtTree?> search({bool dots = false}) async {
+    var foundPaths = await searchPaths();
+    if (foundPaths == null || foundPaths.isEmpty) return null;
+
+    var asciiArtTree = ASCIIArtTree.fromPaths(
+      foundPaths,
+      stripPrefix: stripSearchRoot,
+      style: dots ? ASCIIArtTreeStyle.dots : ASCIIArtTreeStyle.elegant,
+    );
+
+    return asciiArtTree;
+  }
+
+  late PackageConfig _packageConfig;
+  late String _generatedDir;
+
+  /// Performs the imports search and returns the found import paths.
+  Future<List<List<String>>?> searchPaths() async {
+    var currentDir = Directory.current;
+    _packageConfig = (await findPackageConfig(currentDir))!;
+    _generatedDir = p.join('.dart_tool/build/generated');
+
+    var root = from.scheme == 'package' ? _packageConfig.resolve(from)! : from;
+
+    var foundPaths = _searchImportPaths(root, stripSearchRoot: stripSearchRoot);
+    if (foundPaths.isEmpty) {
+      return null;
+    }
+
+    if (!findAll) {
+      foundPaths.sort((a, b) => a.length.compareTo(b.length));
+      var shortest = foundPaths.first;
+      return [shortest];
+    } else {
+      return foundPaths;
+    }
+  }
+
+  List<List<String>> _searchImportPaths(
+    Uri node, {
+    String? stripSearchRoot,
+    Set<Uri>? walked,
+    List<String>? parents,
+    List<List<String>>? found,
+  }) {
+    found ??= [];
+
+    if (walked == null) {
+      walked = {node};
+    } else if (!walked.add(node)) {
+      return found;
+    }
+
+    final nodePath = node.toString();
+    if (parents == null) {
+      parents = [nodePath];
+    } else {
+      parents.add(nodePath);
+    }
+
+    var newImports = _importsFor(node, quiet: quiet)
+        .where((uri) => !walked!.contains(uri))
+        .toList(growable: false);
+
+    for (var import in newImports) {
+      if (_matchesImport(importToFind, import)) {
+        var foundPath = [...parents, import.toString()];
+        found.add(foundPath);
+      } else {
+        _searchImportPaths(import,
+            stripSearchRoot: stripSearchRoot,
+            walked: walked,
+            parents: parents,
+            found: found);
+      }
+    }
+
+    var rm = parents.removeLast();
+    assert(rm == nodePath);
+
+    return found;
+  }
+
+  bool _matchesImport(Object importToFind, Uri import) {
+    if (importToFind is RegExp) {
+      return importToFind.hasMatch(import.toString());
+    } else {
+      return importToFind == import;
+    }
+  }
+
+  List<Uri> _importsFor(Uri uri, {required bool quiet}) {
+    if (uri.scheme == 'dart') return [];
+
+    var filePath = (uri.scheme == 'package' ? _packageConfig.resolve(uri) : uri)
+        ?.toFilePath();
+
+    if (filePath == null) {
+      if (!quiet) {
+        printMessage('» [WARNING] Unable to resolve Uri $uri, skipping it');
```

Comment:
We probably always want to print this warning as it indicates an error.

Reply:
This only happens if the current working directory is not the dart project directory. Try to scan package `foo` from the `import_path` directory:

```shell
cd workspace/import_path

dart run bin/import_path.dart ../foo/web/main.dart dart:html
```

Reply:
It could happen if you haven't ran `pub get` since you added some package to  your pubspec, but also since you described the use case for quiet mode above I think we do want to avoid logging it in this mode.